backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/bottomsheet/BpkModalBottomSheetState.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.bottomsheet
+
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.SheetState
+import androidx.compose.material3.SheetValue
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.remember
+
+@Composable
+@OptIn(ExperimentalMaterial3Api::class)
+fun rememberBpkModalBottomSheetState(
+    skipPartiallyExpanded: Boolean = false,
+    confirmStateChange: (BpkModalBottomSheetValue) -> Boolean = { true },
+): BpkModalBottomSheetState {
+    val delegate = rememberModalBottomSheetState(
+        skipPartiallyExpanded = skipPartiallyExpanded,
+        confirmValueChange = { confirmStateChange(it.toBpkBottomSheetValue()) },
+    )
+    return remember { BpkModalBottomSheetState(delegate) }
+}
+
+enum class BpkModalBottomSheetValue {
+    Collapsed,
+    Expanded,
+    Hidden,
+}
+
+@Stable
+@OptIn(ExperimentalMaterial3Api::class)
+class BpkModalBottomSheetState internal constructor(
```

Comment:
We also need to provide a constructor to hoist the state inside the logic layer. It should mimic the SheetState constructor.

Reply:
Also, can we reuse the state/the value from non-modal version?

Reply:
> We also need to provide a constructor to hoist the state inside the logic layer. It should mimic the SheetState constructor.

Not sure what exactly you mean - I followed the same approach as on the non-modal version

> Also, can we reuse the state/the value from non-modal version?
They have different functionality/values (ie this one has hidden), so no

Reply:
again i followed the same pattern as in the non-modal version - can you clarify what you mean here?